src/main/java/de/retest/recheck/TestCaseFinder.java

```diff
@@ -86,7 +88,41 @@ public static TestCaseFinder getInstance() {
 	 * @return The class name for the test case method in all stack traces.
 	 */
 	public Optional<String> findTestCaseClassNameInStack() {
-		return findTestCaseMethodInStack( toClassName() );
+		final Optional<String> testCaseClassName = findTestCaseMethodInStack( toClassName() );
+		if ( testCaseClassName.isPresent() ) {
+			return testCaseClassName;
+		} else {
+			return findTestCaseClassInStack();
+		}
+	}
+
+	/**
+	 * @return The class name for the test case method in all stack traces.
+	 */
+	public Optional<String> findTestCaseClassInStack() {
+		for ( final StackTraceElement[] stack : Thread.getAllStackTraces().values() ) {
+			final Optional<String> className = findTestCaseClassInStack( stack );
+			if ( className.isPresent() ) {
+				return className;
+			}
+		}
+		return Optional.empty();
+	}
+
+	/**
+	 * @param trace
+	 *            The trace to be used for search.
+	 * @return The class name for the test case method in the given stack trace.
```

Comment:
I assume copy & paste error?

```suggestion
	 * @return The class name for the test being called from.
```